2016-05-01  Atibhav Mittal  <atibhav@lnxsrv02.seas.ucla.edu>

	* timespec.h: Modified comparison(to signed long) of time to generate correct output for sort functions

--- ../coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec.h	2016-05-01 16:25:01.918394000 -0700
@@ -45,8 +45,19 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  signed long int diff = (signed long int)a.tv_sec - (signed long int) b.tv_sec;
+  if(diff > 0)
+    {
+      return 1;
+    }
+  else if (diff == 0)
+    {
+      return 0;
+    }
+  else
+    {
+      return -1;
+    }
 }
 
 # if ! HAVE_DECL_NANOSLEEP
